lib/security_response_header_filter.js

```diff
@@ -109,5 +110,18 @@ module.exports = function addCSPHeaders(hyper, req, next, options) {
         // Finally, assign the csp header variants
         cspHeaderVariants.forEach((name) => { rh[name] = csp; });
         return res;
+    };
+
+    return resPromise.then(addHeaders, (e) => {
+        if (e.constructor.name !== 'HTTPError') {
+            e = new HTTPError({
+                status: 500,
+                body: {
+                    error: e
```

Comment:
This should resemble more the way [hyperswitch does it](https://github.com/wikimedia/hyperswitch/blob/fa0a97c35d62d96f2c6094bcd1503457b0b931ac/lib/server.js#L369-L387)

Reply:
Indeed. I was trying to find that code... 

I will make a patch for HyperSwitch to expose that as a factory method in HTTPError to avoid copy-paste

Reply:
Even better! :+1: